wgpu/tests/shader_primitive_index/mod.rs

```diff
@@ -0,0 +1,253 @@
+            usage: wgpu::BufferUsages::INDEX | wgpu::BufferUsages::COPY_DST,
+        });
+
+    let pipeline = ctx
+        .device
+        .create_render_pipeline(&wgpu::RenderPipelineDescriptor {
+            label: None,
+            layout: None,
+            vertex: wgpu::VertexState {
+                buffers: &[wgpu::VertexBufferLayout {
+                    array_stride: 8,
+                    step_mode: wgpu::VertexStepMode::Vertex,
+                    attributes: &[wgpu::VertexAttribute {
+                        format: wgpu::VertexFormat::Float32x2,
+                        offset: 0,
+                        shader_location: 0,
+                    }],
+                }],
+                entry_point: "vs_main",
+                module: &shader,
+            },
+            primitive: wgpu::PrimitiveState::default(),
+            depth_stencil: None,
+            multisample: wgpu::MultisampleState::default(),
+            fragment: Some(wgpu::FragmentState {
+                entry_point: "fs_main",
+                module: &shader,
+                targets: &[wgpu::ColorTargetState {
+                    format: wgpu::TextureFormat::Rgba8Unorm,
+                    blend: None,
+                    write_mask: wgpu::ColorWrites::ALL,
+                }],
+            }),
+            multiview: None,
+        });
+
+    let width = 2;
+    let height = 2;
+    let texture_size = wgpu::Extent3d {
+        width,
+        height,
+        depth_or_array_layers: 1,
+    };
+    let color_texture = ctx.device.create_texture(&wgpu::TextureDescriptor {
+        label: None,
+        size: texture_size,
+        mip_level_count: 1,
+        sample_count: 1,
+        dimension: wgpu::TextureDimension::D2,
+        format: wgpu::TextureFormat::Rgba8Unorm,
+        usage: wgpu::TextureUsages::RENDER_ATTACHMENT | wgpu::TextureUsages::COPY_SRC,
+    });
+    let color_view = color_texture.create_view(&wgpu::TextureViewDescriptor::default());
+
+    let mut encoder = ctx
+        .device
+        .create_command_encoder(&wgpu::CommandEncoderDescriptor::default());
+
+    let mut rpass = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
+        color_attachments: &[wgpu::RenderPassColorAttachment {
+            ops: wgpu::Operations {
+                load: wgpu::LoadOp::Clear(wgpu::Color::WHITE),
+                store: true,
+            },
+            resolve_target: None,
+            view: &color_view,
+        }],
+        depth_stencil_attachment: None,
+        label: None,
+    });
+
+    rpass.set_pipeline(&pipeline);
+    rpass.set_index_buffer(index_buffer.slice(..), wgpu::IndexFormat::Uint32);
+    rpass.set_vertex_buffer(0, vertex_buffer.slice(..));
+    function(&mut rpass);
+
+    drop(rpass);
+
+    ctx.queue.submit(Some(encoder.finish()));
+
+    let data = capture_rgba_u8_texture(ctx, color_texture, texture_size);
+
+    assert_eq!(data, expected);
+}
+
+fn align_to(n: u32, alignment: u32) -> u32 {
```

Comment:
Is there anywhere else we could share this code with? I feel like we have it in a couple other places in the codebase.

Reply:
I've added changes that move this function to the `wgpu::util` namespace and replaced the occurences I could find. I've also changed the implementation to a modulo operation with an explicit check for zero. That seems to be the more popular version in `wgpu.rs` :-)